addons/hr/models/hr_department.py

```diff
@@ -42,6 +45,13 @@ def _compute_complete_name(self):
             else:
                 department.complete_name = department.name
 
+    @api.depends('parent_path')
+    def _compute_master_department_id(self):
+        # Don't use the cache as the value is updated in SQL
+        parent_path_values = {e['id']: e['parent_path'] for e in self.read(['parent_path'])}
+        for department in self:
+            department.master_department_id = int(parent_path_values[department.id].split('/')[0])
```

Comment:
Could limit to 1 split, as the rest is completely unnecessary since it's all ignored.

Also why not perform this when creating `parent_path_value`? Since we're only reading the paths for the current records it shouldn't be the extra processing it *could* be in other situations (e.g. `read_group`).